test/playlist-maker.test.js

```diff
@@ -838,10 +827,10 @@ QUnit.test('playlist.shuffle() works as expected', function(assert) {
   const list = playlist();
 
   assert.strictEqual(list.length, 4, 'playlist is the correct length');
-  assert.notStrictEqual(list.indexOf(1), -1, '1 is in the list');
-  assert.notStrictEqual(list.indexOf(2), -1, '2 is in the list');
-  assert.notStrictEqual(list.indexOf(3), -1, '3 is in the list');
-  assert.notStrictEqual(list.indexOf(4), -1, '4 is in the list');
+  assert.notStrictEqual(list.findIndex(i => i.originalValue === 1), -1, '1 is in the list');
```

Comment:
Good point. It might be worthwhile to have both assertions (`indexOf` and `findIndex`) to ensure that both are working.

Reply:
Actually, this makes me think, we should probably transform the internal into an array of the originalValues when we return the list. Thoughts?

Reply:
I'm likely missing something here but if we did this how would you determine index of the current item? Wouldn't this bring us back to the original issue?

Reply:
Internally, it would still be the new format and would still allow us to do the correct checks. However, the API currently says that you get what you input, we should map and return the originalValues if they exist when returning the list.
If doing so causes things not to work then it means we need to think of whether we need to make a breaking change.

Reply:
Unless we kept two versions of list — one internal with the new format and another one that is returned publicly — the approach you described will not work since the `originalValues` won't have the `playlistItemId_` property based on which the index of the item in the list array is determined. Having two lists, however,  might be a bit messy. What are your thoughts? @gkatsev

Reply:
Why does the return value matter for `playlistItemId_`?

Reply:
Ignore above, was thinking about something entirely different. You're right `playlistItemId_` shouldn't really matter when returning list.

Reply:
@gkatsev ping